doc/reference/tooling/tt_cli/replicaset.rst

```diff
@@ -62,42 +62,170 @@ For a replica outside the current ``tt`` environment, specify its URI and access
-~~~~~~~~~~~~~~
+.. _tt-replicaset-promote:
 
-Use one of the following ways to pass the credentials of a Tarantool user when
-connecting to the instance:
+promote
+-------
 
-*   The ``-u`` (``--username``) and ``-p`` (``--password``) options:
+..  code-block:: console
 
-    ..  code-block:: console
+    $ tt replicaset promote {APPLICATION:APP_INSTANCE | URI} [OPTIONS ...]
+    # or
+    $ tt rs promote {APPLICATION:APP_INSTANCE | URI} [OPTIONS ...]
 
-        $ tt replicaset status 192.168.10.10:3301 -u myuser -p p4$$w0rD
+``tt replicaset promote`` (``tt rs promote``) promotes the specified instance,
+making it a leader of its replica set.
+This command works on Tarantool clusters with a local YAML
+configuration and Cartridge clusters.
 
-*   The connection string:
+.. note::
 
-    ..  code-block:: console
+    To promote an instance in a Tarantool cluster with a :ref:`centralized configuration <configuration_etcd>`,
+    use :ref:`tt cluster replicaset promote <tt-cluster-replicaset-promote>`.
 
-        $ tt replicaset status myuser:p4$$w0rD@192.168.10.10:3301
+.. _tt-replicaset-promote-config:
 
-*   Environment variables ``TT_CLI_USERNAME`` and ``TT_CLI_PASSWORD``:
+Promoting in clusters with local YAML configurations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    ..  code-block:: console
+``tt replicaset promote`` works on Tarantool clusters with local YAML configurations
+with :ref:`failover modes <configuration_reference_replication_failover>`
+``off``, ``manual``, and ``election``.
 
-        $ export TT_CLI_USERNAME=myuser
-        $ export TT_CLI_PASSWORD=p4$$w0rD
-        $ tt replicaset status 192.168.10.10:3301
+In failover modes ``off`` or ``manual``, this command updates the cluster
+configuration file according to the specified arguments and reloads it:
+
+-   ``off`` failover mode: the command sets :ref:`database.mode <configuration_reference_database_mode>`
+    to ``rw`` on the specified instance.
+
+    .. important::
+
+        If failover is ``off``, the command doesn't consider the modes of other
+        replica set members, so there can be any number of read-write instances in one replica set.
+
+-   ``manual`` failover mode: the command updates the :ref:`leader <configuration_reference_replicasets_name_leader>`
+    option of the replica set configuration. Other instances of this replica set become read-only.
+
+Example:
+
+..  code-block:: console
+
+    $ tt replicaset promote my-app:storage-001-a
+
+If some members of the affected replica set are running outside the current ``tt``
+environment, ``tt replicaset promote`` can't ensure the configuration reload on
+them and reports an error. You can skip this check by adding the ``-f``/``--force`` option:
+
+..  code-block:: console
+
+    $ tt replicaset promote my-app:storage-001-a --force
+
+In the ``election`` failover mode, ``tt replicaset promote`` initiates the new leader
+election by calling :ref:`box_ctl-promote` on the specified instance. The
```

Comment:
I've separated it because of conceptually different action: in this case, the command doesn't write the cluster config (unlike with failover `off`  or `manual`). Also, the tt environment doesn't matter in this case (the paragraph above).